### ISSUE TITLE
Public shares: Wrong html 'lang' attribute

### ISSUE THREAD

**Issue by jasperw1996** [owncloud/web#12409]
### Describe the bug

If I share a folder via public link and someone opens it, the language of the sharing page will be set according to the browser's preference (the `Accept-Language` header). But at the same time, the html `lang` attribute is always set to `en`. This could lead to strange phenomenons: e. g. a screen reader reading German text with a strong English accent – because it relies on a `lang` attribute which doesn't fit the actual language.

### Steps to reproduce

1. Share a folder or file via public link
2. Access the share in a different browser where you're not logged in (set the browser's preferred language to something other than English)
3. Open developer tools to see that the `lang` attribute and the actual language of the page mismatch:
![Image](https://github.com/user-attachments/assets/ce10cdc2-f5d1-4be8-ad76-3e036da52965)

### Expected behavior

The actual language of the page should always change together with the html `lang` global attribute. This is working fine while being logged in and changing the language manually in the settings, … 

### Actual behavior

… but it doesn't work with public shares – and people who are not logged in.

### Setup

```markdown
- Web version: 11.3.1
- oCIS version: 7.1.2 Community
- browser: Google Chrome 134.0.6998.165
```

<details>
<p>

```
podman run --rm -it -v ocis-config:/etc/ocis/ docker.io/owncloud/ocis:7 init
podman run --rm -p 9200:9200 -v ocis-config:/etc/ocis -v ocis-data:/var/lib/ocis -e IDM_CREATE_DEMO_USERS=true   docker.io/owncloud/ocis:7
```

</p>
</details>

### Additional context

This might relate to https://github.com/owncloud/web/issues/9288

**Comment by jasperw1996** [comment]
Nice job, thank you very much! :)

The lang attribute is now adapting to the browser's preferred language.
This might be pedantic - but now there is a small glitch vice versa: If I set the browser's preferred language to something that is not supported by OCIS (e. g. "Luxembourgish"), the page will fall back to English, while the lang attribute is set to "lb". 

![Image](https://github.com/user-attachments/assets/30c819e8-9332-4f72-a7c1-58f4efb62038)

So maybe in this special case the lang attribute should have some sort of fallback, too?

**Comment by mzner** [comment]
> Nice job, thank you very much! :)
> 
> The lang attribute is now adapting to the browser's preferred language. This might be pedantic - but now there is a small glitch vice versa: If I set the browser's preferred language to something that is not supported by OCIS (e. g. "Luxembourgish"), the page will fall back to English, while the lang attribute is set to "lb".
> 
> ![Image](https://github.com/user-attachments/assets/30c819e8-9332-4f72-a7c1-58f4efb62038)
> 
> So maybe in this special case the lang attribute should have some sort of fallback, too?

Good point! I will check it out again and see what can be done about it.

**Comment by mzner** [comment]
> > Nice job, thank you very much! :)
> > The lang attribute is now adapting to the browser's preferred language. This might be pedantic - but now there is a small glitch vice versa: If I set the browser's preferred language to something that is not supported by OCIS (e. g. "Luxembourgish"), the page will fall back to English, while the lang attribute is set to "lb".
> > ![Image](https://github.com/user-attachments/assets/30c819e8-9332-4f72-a7c1-58f4efb62038)
> > So maybe in this special case the lang attribute should have some sort of fallback, too?
> 
> Good point! I will check it out again and see what can be done about it.

@jasperw1996 We have added a fix to the mismatch when browser's language is set to a language that is not supported by us.

https://github.com/owncloud/web/issues/12486

**Comment by jasperw1996** [comment]
Nice! Thanks alot :)
